pkg/stanza/fileconsumer/file.go

```diff
@@ -73,7 +80,9 @@ func (m *Manager) Stop() error {
 		m.cancel = nil
 	}
 	m.wg.Wait()
-	m.telemetryBuilder.FileconsumerOpenFiles.Add(context.TODO(), int64(0-m.tracker.ClosePreviousFiles()))
+	if m.tracker != nil {
```

Comment:
As it turns out, we don't need to pass persister into tracker. 

I've introduced a new method `GetArchiveMetadata`, which returns the metadata to be stored in archive (i.e. `knownFiles[2]`). It looks much simpler now. PTAL!

Reply:
Since we initialize the tracker in `Start`, the component lifecycle tests for `otlpjsonfilereceiver` are causing panics due to `Shutdown` being called first. We need to implement a check for this.

Reply:
Same reason as we check for `m.cancel` here
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/985dddba71e26212e7e0535965fbec205135958f/pkg/stanza/fileconsumer/file.go#L71-L74